core-primitives/stf-state-handler/src/in_memory_state_file_io.rs

```diff
@@ -371,7 +415,12 @@ mod tests {
 	}
 
 	fn create_in_memory_state_file_io(shards: &[ShardIdentifier]) -> TestStateFileIo {
-		InMemoryStateFileIo::new(DefaultHasher::default(), shards)
+		InMemoryStateFileIo::new(
+			shards,
+			Box::new(|x| x),
+			Box::new(|| TestState::default()),
+			Box::new(|x| x),
+		)
```

Comment:
I can't understand why we have boxes here. :)

Reply:
In order to take an arbitrary function object as member, I believe that's the way to go? Or do you know of another way? 

In any case, this is a test implementation only, it's not used in production code.

Reply:
I think it would be more rust like to have a plain struct as a member that implements a trait, which defines these functions. Then you could just that struct as phantom data.

Reply:
Granted, that would be an option 👍 

The reason why I didn't like it in this case, is that is adds a lot of boilerplate code for just a simple function interface. In this case 3 traits and `n*3` struct implementations (`n` being the number of use cases).

Reply:
After discussing further, we decided to leave it as is for now, since this part of the code will most likely get reworked anyway with the upcoming PR for state handling (issue #459)